### PR TITLE
Add data for HTMLSelectElement showPicker

### PR DESCRIPTION
**Pull request by lukewarlow**
#### Summary

Add data for HTMLSelectElement showPicker

#### Test results and supporting details

Pull request for html spec: https://github.com/whatwg/html/pull/9754

Chrome Status Entry: https://chromestatus.com/feature/5111537299881984

Content PR: https://github.com/mdn/content/pull/29406

### CODE REVIEW

**Comment by queengooborg** [comment]
I noticed this was marked as a draft, but this looks ready to go?

**Comment by lukewarlow** [comment]
@queengooborg I was waiting till the spec PR got merged. Currently the spec link won't work. Happy to unmark as draft if you think that's fine

**Comment by queengooborg** [comment]
Ah, thanks for pointing that out!  If you'd like, we can mark this as non-standard for now, and then follow up with another PR to add the spec URL when it becomes standard; or, we can leave this unmerged until the spec changes are official.  I have no preference either way!

**Comment by lukewarlow** [comment]
@queengooborg the spec PR just got merged so this is ready to go.

**Review by Elchi3 — COMMENTED** [review]
The Firefox bug is marked as fixed in 121. https://bugzilla.mozilla.org/show_bug.cgi?id=1854112 and the Chromestatus says shipping in 121.

Can you update this PR?

**Comment by lukewarlow** [comment]
Added Firefox flagged support and marked as stable in chrome from 121 (here's the PR to flip it on waiting for review https://chromium-review.googlesource.com/c/chromium/src/+/5034127)

**Review by Elchi3 — APPROVED** [review]
Thanks! 👍

**Comment by lukewarlow** [comment]
(sorry last minute change realised it's also out in Safari Tech Preview)

File: api/HTMLSelectElement.json [hunk 1]
```diff
@@ -843,6 +843,54 @@
...
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLSelectElement/showPicker",
+          "spec_url": "https://html.spec.whatwg.org/multipage/input.html#dom-select-showpicker",
+          "support": {
+            "chrome": {
+              "version_added": "121"
+            },
+            "chrome_android": "mirror",
+            "edge": "mirror",
+            "firefox": {
```
Comment (hamishwillee) [inline api/HTMLSelectElement.json:856]: FYI only the bug now says this is enabled in "nightly" so have updated this to `preview` in https://github.com/mdn/browser-compat-data/pull/21280

**Pull request merged**